src/modules/Dropdown/Dropdown.js

```diff
@@ -1167,6 +1169,7 @@ export default class Dropdown extends Component {
       onClick: this.handleItemClick,
       selected: selectedIndex === i,
       ...opt,
+      key: getKeyOrValue(opt.key, opt.value),
```

Comment:
## Problem

Before #1639, `key` [was captured](https://github.com/Semantic-Org/Semantic-UI-React/blob/v0.68.3/src/modules/Dropdown/Dropdown.js#L1155) from `value`. In #1639 we refactored this to factory usage, but as we [know](https://github.com/Semantic-Org/Semantic-UI-React/issues/1710#issuecomment-304513318) factory will not create `key` automatically when it recieves an object and it's expected behaviour.

But, it's **breaking** change that I think should be reverted, because our users rely on that `key` can be automatically generated from a `value` prop (#1710).